packages/sw-lib/src/lib/sw-lib.js

```diff
@@ -39,14 +39,29 @@ class SWLib {
    * localhost.
    * @param {boolean} clientsClaim To claim currently open clients set
    * this value to true. (Default false).
+   * @param  {String} [input.directoryIndex]  The directoryIndex will
+   * check cache entries for a URLs ending with '/' to see if there is a hit
+   * when appending the directoryIndex (i.e. '/index.html').
    */
-  constructor({cacheId, clientsClaim} = {}) {
+  constructor({cacheId, clientsClaim, directoryIndex} = {}) {
```

Comment:
Actually, can you get away with using [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) here?

`directoryIndex = 'index.html'` seems like it would be sufficient and is easier to read.